astropix.py

```diff
@@ -111,46 +130,53 @@ def asic_init(self, dac_setup: dict = None, bias_setup:dict = None, digital_mask
         self._make_digitalconfig()
         self._make_reconfig()
         # Loads it to the chip
-        print("LOADING TO ASIC...")
+        logger.info("LOADING TO ASIC...")
         self.asic_update()
-        print("ASIC SUCCESSFULLY CONFIGURED")
+        logger.info("ASIC SUCCESSFULLY CONFIGURED")
+
 
     # The method to write data to the asic. Called whenever somthing is changed
     # or after a group of changes are done. Taken straight from asic.py.
-    # Might need updating down the line but it shoudl still work
-
+    # Might need updating down the line but it should still work
     def asic_update(self):
-        """Update ASIC"""
+        """
+        Remakes configbits and writes to asic. 
+        Takes no input and does not return
+        """
 
-        # Not needed for v2
-        # dummybits = self.gen_asic_pattern(BitArray(uint=0, length=245), True)
-        # Write config
         asicbits = self.nexys.gen_asic_pattern(self._construct_asic_vector(), True)
         self.nexys.write(asicbits)
 
 
     # Methods to update the internal variables. Please don't do it manually
     # This updates the dac config
-    def update_dac(self, dac_config:dict, update_now: bool = True):
-        if self._asic_start:
-            self.dac_setup.update(dac_config)
-            # This will automatically load the new configuration if needed
-            if update_now: self.asic_update()
-        else: raise Exception("asic_init must first be called!")
-
-    def update_bias(self, bias_cfg:dict, update_now: bool = True):
-        if self._asic_start:
-            self.bias_setup.update(bias_cfg)
-        else: raise Exception("asic_init must first be called!")
-
-    # This makes a new digital mask and rewrites the configuration to the asic 
-    def update_mask_digit(self, maskstr: str):
-        if self._asic_start:
-            self._make_digital_mask(maskstr)
-            self.asic_update()    
-        else: raise Exception("asic_init must first be called!")
+    def update_asic_config(self, bias_cfg:dict = None, dac_cfg:dict = None, maskstr:str = None):
+        """
+        Updates and writes confgbits to asic
+
+        bias_cfg:dict - Updates the bias settings. Only needs key/value pairs which need updated
+        dac_cfg:dict - Updates DAC settings. Only needs key/value pairs which need updated
+        """
+        if self.asic_start:
+            if bias_cfg is not None:
+                self.bias_setup.update(bias_cfg)
+            if dac_cfg is not None:
+                self.dac_setup.update(dac_cfg)
+            if maskstr is not None:
+                self._make_digital_mask(maskstr)
+            else: logger.info("Nothing to do")
+            
+            self.asic_update()
+        else: raise RuntimeError("Asic has not been initalized")
```

Comment:
I set these as runtime errors since they are triggered if something is run out of order which could lead to objects which haven't been initialized yet being called which would raise a NameError, but it would be a lot harder to track down than this.